pkg/apis/resource/validation/validation.go

```diff
@@ -119,10 +119,41 @@ func validateDeviceClaim(deviceClaim *resource.DeviceClaim, fldPath *field.Path,
 	return allErrs
 }
 
-func gatherRequestNames(deviceClaim *resource.DeviceClaim) sets.Set[string] {
-	requestNames := sets.New[string]()
+type requestNames map[string]sets.Set[string]
+
+func (r requestNames) Has(s string) bool {
+	segments := strings.Split(s, "/")
+	// If there are more than one / in the string, we
+	// know there can't be any match.
+	if len(segments) > 2 {
+		return false
+	}
+	// If the first segment doesn't have a match, we
+	// don't need to check the other one.
+	subRequestNames, found := r[segments[0]]
+	if !found {
+		return false
+	}
+	if len(segments) == 1 {
+		return true
+	}
```

Comment:
According to this code, a reference to a request is valid even if that request has subrequests. Otherwise this code probably would have to check that `subRequestNames` is nil before returning true.